File: GXpert/GXpert.Web/Modules/Institute/InstituteTeacher/InstituteTeacherColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.InstituteTeacher")]
[BasedOnRow(typeof(InstituteTeacherRow), CheckNames = true)]
public class InstituteTeacherColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string TeacherPrn { get; set; }
    public string InstituteName { get; set; }
    [QuickFilter]
    public string ClassTitle { get; set; }
    [QuickFilter]
    public string SubjectTitle { get; set; }
    [EditLink]
    public string Description { get; set; }
    public string AcademicYearName { get; set; }
     public string Semester { get; set; }
    public string InstituteDivision{ get; set; }


}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteTeacherColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { InstituteTeacherRow } from "./InstituteTeacherRow";

export interface InstituteTeacherColumns {
    Id: Column<InstituteTeacherRow>;
    TeacherPrn: Column<InstituteTeacherRow>;
    InstituteName: Column<InstituteTeacherRow>;
    ClassTitle: Column<InstituteTeacherRow>;
    SubjectTitle: Column<InstituteTeacherRow>;
    Description: Column<InstituteTeacherRow>;
    AcademicYearName: Column<InstituteTeacherRow>;
    Semester: Column<InstituteTeacherRow>;
    InstituteDivision: Column<InstituteTeacherRow>;
}

export class InstituteTeacherColumns extends ColumnsBase<InstituteTeacherRow> {
    static readonly columnsKey = 'Institute.InstituteTeacher';
    static readonly Fields = fieldsProxy<InstituteTeacherColumns>();
}